### ISSUE TITLE
needless_question_mark follows macros and suggests to replace code inside the macro definition

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6921]
Lint name: needless_question_mark

I tried this code:

```rust
macro_rules! try_ {
    ($expr:expr) => {
        || -> _ { Some($expr) }()
    };
}

fn main() {
    let x = Some(3);
    let x = try_!(x?);
}
```
Suggestion:

````
warning: question mark operator is useless here
 --> src/main.rs:3:19
  |
3 |         || -> _ { Some($expr) }()
  |                   ^^^^^^^^^^^ help: try: `x`
...
9 |     let x = try_!(x?);
  |             --------- in this macro invocation
  |
  = note: `#[warn(clippy::needless_question_mark)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_question_mark
  = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
````

We can't just fill in the macro arguments into the macro definition... :upside_down_face: 
I guess the easiest way is to just ignore macros?

### Meta
`clippy 0.1.52 (107896c 2021-03-15)`
